proposals/texture-buffers.md

```diff
@@ -0,0 +1,414 @@
+| _C_ is `i32` or `u32`<br/>_CF_ depends on the storage texel format _F_. [See the texel format table](https://w3.org/TR/WGSL/#storage-texel-formats) for the mapping of texel format to channel format. | `@must_use fn textureStore(t : texture_buffer<F, read_write>, coords : C, value: vec4<CF>)` |
+
+
+# API
+
+
+## Limits
+
+| Limit name               | Type        | Limit class | Default                   |
+| ------------------------ | ----------- | ----------- | ------------------------- |
+| **maxTextureBufferSize** | `GPUSize64` | maximum     | 134217728 bytes (128 MiB) |
+
+
+## Adapter Capability Guarantees
+
+Add "`maxTextureBufferSize` must be <= `maxBufferSize`".
+
+
+## GPUFeatureName
+
+Add `"texture-buffers"` as a new enum entry.
+
+
+## Resource Usages
+
+[[Modify description of internal usages]]
+
+**storage**<br/>
+Read/write storage resource binding. Allowed by buffer `STORAGE`, texture `STORAGE_BINDING`, or `TEXTURE_BUFFER`.
+
+**storage-read**<br/>
+Read-only storage resource bindings. Preserves the contents. Allowed by buffer `STORAGE`, texture `STORAGE_BINDING`, or `TEXTURE_BUFFER`.
+
+
+## Buffer Usages
+
+[[Add new const to `GPUBufferUsage` namespace]]
+
+```javascript
+  const GPUFlagsConstant TEXTURE_BUFFER = 0x0400;
+```
+
+
+## GPUTextureBufferView
+
+[[New subsection of **Textures and Texture Views**]]
+
+A `GPUTextureBufferView` is a view onto some subset of the buffer subresources defined by a particular `GPUBuffer`.
+
+```javascript
+[Exposed=(Window, Worker), SecureContext]
+interface GPUTextureBufferView {
+};
+GPUTextureBufferView includes GPUObjectBase;
+```
+
+`GPUTextureBufferView` has the following immutable properties:
+
+> **[[buffer]], readonly**<br/>
+> The `GPUBuffer` into which this is a view.
+>
+> **[[descriptor]], readonly**<br/>
+> The `GPUTextureBufferViewDescriptor` describing this texture buffer view.
+>
+> All optional fields of `GPUTextureBufferViewDescriptor` are defined.
+
+
+### Texture Buffer View Creation
+
+```javascript
+dictionary GPUTextureBufferViewDescriptor : GPUObjectDescriptorBase {
+    GPUTextureFormat format;
+    GPUSize64 offset = 0;
+    GPUSize64 size;
+};
+```
+
+`GPUTextureBufferViewDescriptor` has the following members:
+
+> **format, of type GPUTextureFormat**<br/>
+> The format of the texture buffer view.
+>
+> **offset, of type GPUSize64, defaulting to 0**<br/>
+> The offset, in bytes, from the beginning of the buffer to the range exposed by the texture buffer view.
+>
+> **size, of type GPUSize64**<br/>
+> The size, in bytes, of the texture buffer view. If not provided, specifies the range starting at `offset` and ending at the end of the buffer.
+
+**createView(descriptor)**<br/>
```

Comment:
nit: Do we expect any other view creation in the future? It might be nicer to make it clear in the name that this is just for TextureBuffer (but also `createTextureView` would be weird in a buffer, so maybe `TexelBuffer` still makes more sense as a name?)

Reply:
Happy to rename to `createTextureBufferView` if more explicitness is desired.

I have a very slight preference for "texture buffer" over "texel buffer" for this feature in general, but I don't feel that strongly about it.

Reply:
Given that these are not really textures (they are not a new kind of texture as currently defined in the spec) it might be worth disambiguating and using `TexelBuffer` instead.

Reply:
Renamed to texel buffer as decided at the F2F.